src/solver/contribute_springs_to_sparse_matrix.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <KokkosSparse.hpp>
+#include <Kokkos_Core.hpp>
+
+#include "src/dof_management/freedom_signature.hpp"
+
+namespace openturbine {
+template <typename CrsMatrixType>
+struct ContributeSpringsToSparseMatrix {
+    using RowDataType = typename CrsMatrixType::values_type::non_const_type;
+    using ColIdxType = typename CrsMatrixType::staticcrsgraph_type::entries_type::non_const_type;
+    Kokkos::View<FreedomSignature* [2]> element_freedom_signature;
+    Kokkos::View<size_t* [2][3]> element_freedom_table;
+    Kokkos::View<double* [2][2][3][3]>::const_type dense;  //< Element Stiffness matrices
+    CrsMatrixType sparse;                                  //< Global sparse stiffness matrix
+
+    KOKKOS_FUNCTION
+    void operator()(size_t i_elem) const {
+        constexpr auto is_sorted = true;
+        constexpr auto force_atomic = false;
```

Comment:
This need to have logic to be true when we're running in parallel.  Two springs can share a single node, so it will lead to a race condition if not atomic

Reply:
I changed both sparse_matrix and vector contribution functions with `force_atomic` logic to avoid the race condition issue. Could you let me know if they look good?

Reply:
Also any ideas on how to unit test these solver functions?

Reply:
That logic for the atomic should work perfectly. 

Unit testing these functions isn't hard, but it does require setting up a the element freedom table, sparse matrix row pointers, and sparse matrix column indices manually.  This is mostly just a lot of typing, since our matrices are pretty simple. You can see something similar in the tests I just merged in for constructing the k matrix column indices.